enhancements/cli_manager.md

```diff
@@ -0,0 +1,144 @@
+---
+title: cli-manager
+authors:
+  - "@sallyom"
+reviewers:
+  - TBD
+  - "@soltysh"
+approvers:
+  - TBD
+  - "@soltysh"
+creation-date: 2019-12-03
+last-updated: 2019-12-03
+status: provisional
+---
+
+# cli-manager
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal is for a component to manage various tools available for OpenShift and related services.
+A CLI manager is desired so that a user can discover, install and upgrade tools easily and from a single
+location.  Each component is responsible for serving its files and registering information regarding supplied binaries
+to a central index location.  The CLI manager will retrieve all the binaries from the index location.
+Additionally, CLI manager must be supported in disconnected environments.  Currently, we provide disconnected environments 
+cli-artifacts and oc download links that do not require anything outside the cluster.  The goal is to provide the same for other
+binaries/tools/CLIs.  The CLI manager will provide a mechanism to provide, list, install, and upgrade OpenShift tools/plugins/CLIs from
+within a cluster.  The [ConsoleCLIDownload CRD](https://github.com/openshift/api/blob/master/console/v1/0000_10_consoleclidownload.crd.yaml) 
+may be extended to meet the requirements of a new CLI manager.
+
+## Motivation
+
+As more services are created on top of OpenShift, more CLIs are introduced to simplify interaction with these services.
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist, 
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  We need a mechanism for providing and consuming tools that is simple to add on to as new CLIs are 
+developed from a variety of sources - and this needs to be managed from within a cluster.
+This is the motivation for the CLI manager.
+
+### Goals
+
+Each component wishing to provide customers with their binaries will build and publish them via an official channel to an
+index location from which the CLI manager will obtain the files.
+
+Two routes for index:    
+1.  Central index that has links to where artifacts are stored
+2.  Central repository where all artifacts are stored 
```

Comment:
Isn't this the existing mirror.openshift?
https://mirror.openshift.com/pub/openshift-v4/clients/

CLIs are already being published there today by the CLI creators.

Reply:
I added that info.